app/packages/core/src/plugins/SchemaIO/components/DynamicIO.tsx

```diff
@@ -19,9 +19,10 @@ export default function DynamicIO(props) {
   // todo: need to improve initializing default value in state
   useEffect(() => {
     if (
-      isPrimitiveType(type) &&
-      data !== defaultValue &&
-      !isPathUserChanged(path)
+      !isCompositeView(schema) &&
+      !isEqual(data, defaultValue) &&
+      !isPathUserChanged(path) &&
+      !isNullish(defaultValue)
```

Comment:
`onChange` should never be called if `defaultValue` is nullish?

Reply:
This `onChange` is only used to initialize the default value (i.e. when component initially mount of when default value in the schema is updated externally). When a user makes a change, this flow is not used.